lib/components/Masthead/Masthead.tsx

```diff
@@ -2,236 +2,195 @@ import * as React from 'react';
-        return <ul role='menu' id='app-menu'>
-            <li key='user-label' className={cx('masthead-toolbar-menu-item', 'tall', 'inline-text-overflow')}>
-                <UserLabel {...this.props.user} />
-            </li>
-            {
-                this.props.mobileMenuItems &&
-                this.props.mobileMenuItems.map(action => {
-                    switch (action.type) {
-                        case 'link':
-                            return <li
-                                    className={cx('masthead-toolbar-menu-item', 'inline-text-overflow')}
-                                    key={action.reactKey}
-                                >
-                                    <a role='menuitem' href={action.href}>
-                                        {action.menuLabel}
-                                    </a>
-                                </li>;
-                        case 'button':
-                            return <li
-                                    className={cx('masthead-toolbar-menu-item', 'inline-text-overflow')}
-                                    key={action.reactKey}
-                                >
-                                    <button className={cx('inline-text-overflow')} role='menuitem' onClick={action.onClick}>
-                                        {action.menuLabel}
-                                    </button>
-                                </li>;
-                        case 'submenu':
-                            return <li
-                                    className={cx('masthead-toolbar-menu-item', 'tall', 'sub-menu')}
-                                    key={action.reactKey}
-                                >
-                                    <Accordion
-                                        attr={{
-                                            ariaRole: 'menuitem',
-                                            dataTestHook: action.dataTestHook
-                                        }}
-                                        expanded={action.expanded}
-                                        id={action.dataTestHook}
-                                        label={action.menuLabel}
-                                        onToggle={action.onToggle}
-                                    >
-                                        {action.content}
-                                    </Accordion>
-                                </li>;
+    private toolbarItems: { [key: string]: ToolbarItem } = {
+        'settings': {
+            icon: 'settings',
+            contentPanel: {
+                title: this.props.toolBarItems.settings.title,
+                content: this.props.toolBarItems.settings.content,
+                actions: {
+                    cancel: {
+                        event: this.props.toolBarItems.settings.actions.cancel.event ? this.props.toolBarItems.settings.actions.cancel.event : () => this.togglePanel('settings'),

```

Comment:
i think this.togglePanel('settings') should get fired regardless of what they pass in, maybe just call both

Reply:
sounds reasonable